vpp/vaapipostprocess_scaler.cpp

```diff
@@ -249,6 +328,61 @@ YamiStatus VaapiPostProcessScaler::setDeinterlaceParam(const VPPDeinterlaceParam
     return createDeinterlaceFilter(deinterlace);
 }
 
+YamiStatus VaapiPostProcessScaler::setColorBalanceParam(const VPPColorBalanceParameters& colorbalance)
+{
+    ColorBalanceList& supportedModeRanges = m_brightness.supportedModeRanges;
+    ColorBalanceModeRange modeRange;
+    ColorBalanceParams* param;
+
+    VAProcFilterCapColorBalance caps[VAProcColorBalanceCount];
+    if (supportedModeRanges.empty()) {
+        //query from libva.
+        uint32_t num = VAProcColorBalanceCount;
+        YamiStatus status = queryVideoProcFilterCaps(VAProcFilterColorBalance, caps, &num);
+        if (status != YAMI_SUCCESS)
+            return status;
+        for (uint32_t i = 0; i < num; i++) {
+            modeRange.mode = mapToVppColorBalanceMode(caps[i].type);
+            modeRange.range = caps[i].range;
+            supportedModeRanges.insert(modeRange);
+        }
+    }
+    if (colorbalance.mode == COLORBALANCE_NONE) {
+        m_brightness.filter.reset();
```

Comment:
thanks @xuguangxin for your constructive suggestion.
How about to reset all the colorbalance filters, such as hue, contrast, brightness and saturation,  when getting COLORBALANCE_NONE ?